bigquery/google/cloud/bigquery/client.py

```diff
@@ -1548,6 +1557,14 @@ def load_table_from_dataframe(
                         PendingDeprecationWarning,
                         stacklevel=2,
                     )
+                else:
+                    warnings.warn(
```

Comment:
> If the warning should be removed altogether in this PR, please let me know.

Yes, let's wait until #8140, #8142, and #9044 are closed before adding this warning.